src/Simplify_Stmts.cpp

```diff
@@ -121,6 +127,13 @@ Stmt Simplify::visit(const IfThenElse *op) {
                equal(unwrapped_condition, 0 < then_for->extent)) {
         // This guard is redundant
         return then_case;
+    } else if (then_if &&
```

Comment:
This is a sort of distributive law for ifs? Would maybe help to add a comment:
```
Before:
if (a) {
  if (b) X
} else {
  if (b) Y
}

After:
if (b) {
  if (a) X
  else Y
}
```

I think it requires the conditions a, b to be pure. If they have side-effects then you're changing which ones get evaluated when.